File: AngularEMS-main/AngularEMS-main/src/app/models/employee.ts
import { Department } from "./department";
import { Gender } from "./gender";

export class Employee {
    id!: number;
    name!: string;
    dateOfBirth!: Date;
    gender!: Gender;
    email!: string;
    mobileNo!: number;
    departmentId!: number;
    department!: Department;
}

File: EMS_API-main/EMS_API-main/EmsApi/Models/Employee.cs
using System.ComponentModel.DataAnnotations;

namespace EmsApi.Models
{
    public class Employee
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        public DateTime DateOfBirth { get; set; }
        public Gender Gender { get; set; }
        [Required]
        [EmailAddress]
        public string Email { get; set; }
        [RegularExpression(@"^\d{10}$")]
        public long MobileNo { get; set; }
        public int DepartmentId { get; set; }
        public virtual Department? Department { get; set; }
    }
}
